code/modules/clothing/suits/hoodies.dm

```diff
@@ -24,10 +26,14 @@
 	..()
 
 /obj/item/clothing/suit/storage/hooded/proc/RemoveHood()
-	icon_state = initial(icon_state)
-	item_state = initial(item_state)
-	suittoggled = 0
-
+	if(!lasticonstate && !lastitemstate)
+		suittoggled = 0
+	else
+		icon_state = "[lasticonstate]"
+		item_state = "[lastitemstate]"
+		suittoggled = 0
```

Comment:
You only need one of these vars because item_state should be the same as icon_state in basically all cases. Also, use TRUE/FALSE instead of 1/0 if they are booleans, please.